next/tp6/env_pendulum.py

```diff
@@ -1,142 +1,186 @@
+        self.NDT = 5  # number of euler integration step per environment step
+        # (i.e integration interval is DT/NDT)
+
+        self.Kf = 1.0  # Friction coefficient
+
+        self.costWeights = {"q": 1, "v": 1e-1, "u": 1e-3, "tip": 0.0}
         self.tipDes = float(nbJoint)
-        
-    def cost(self,x=None,u=None):
-        if x is None: x = self.x
-        cost  = 0.
-        q,v = x[:self.nq],x[-self.nv:]
-        qdes = self.xdes[:self.nq]
-        cost += self.costWeights['q']*np.sum((q-qdes)**2)
-        cost += self.costWeights['v']*np.sum(v**2)
-        cost += 0 if u is None else self.costWeights['u']*np.sum(u**2)
-        cost += self.costWeights['tip']*(self.tip(q)-self.tipDes)**2
+
+    def cost(self, x=None, u=None):
+        if x is None:
+            x = self.x
+        cost = 0.0
+        q, v = x[: self.nq], x[-self.nv :]
+        qdes = self.xdes[: self.nq]
+        cost += self.costWeights["q"] * np.sum((q - qdes) ** 2)
+        cost += self.costWeights["v"] * np.sum(v**2)
+        cost += 0 if u is None else self.costWeights["u"] * np.sum(u**2)
+        cost += self.costWeights["tip"] * (self.tip(q) - self.tipDes) ** 2
         return cost
 
-    def tip(self,q=None):
-        '''Return the altitude of pendulum tip'''
-        if q is None: q = self.x[:self.nq]
-        pin.framesForwardKinematics(self.rmodel,self.rdata,q)
+    def tip(self, q=None):
+        """Return the altitude of pendulum tip"""
+        if q is None:
+            q = self.x[: self.nq]
+        pin.framesForwardKinematics(self.rmodel, self.rdata, q)
         return self.rdata.oMf[1].translation[2]
 
     def jupyter_cell(self):
         return self.viewer.viewer.jupyter_cell()
-    
-# --- SPIN-OFF ----------------------------------------------------------------------------------
-# --- SPIN-OFF ----------------------------------------------------------------------------------
-# --- SPIN-OFF ----------------------------------------------------------------------------------
+
+
+# --- SPIN-OFF ----------------------------------------------------------------
+# --- SPIN-OFF ----------------------------------------------------------------
+# --- SPIN-OFF ----------------------------------------------------------------
+
 
 class EnvPendulumDiscrete(env_abstract.EnvDiscretized):
-    def __init__(self,nbJoint=1,**kwargs):
-        env = EnvPendulum(nbJoint,**kwargs)
-        env.DT=5e-1     # Larger integration step to allow larger discretization grid
-        env.Kf=0.1      # Reduced friction, because larger steps would make friction unstable.
-        env_abstract.EnvDiscretized.__init__(self,env,21,11)
-        self.discretize_x.modulo = np.pi*2
+    def __init__(self, nbJoint=1, **kwargs):
+        env = EnvPendulum(nbJoint, **kwargs)
+        env.DT = 5e-1  # Larger integration step to allow larger discretization grid
+        env.Kf = (
+            0.1  # Reduced friction, because larger steps would make friction unstable.
+        )
```

Comment:
```suggestion
        env.Kf = 0.1  # Reduced friction, because larger steps would make friction unstable.
```


if this is less than 88 columns. Otherwise we can put the comment its own line, above.